libs/testproxy/server.go

```diff
@@ -0,0 +1,177 @@
+package testproxy
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/databricks/cli/internal/testutil"
+	"github.com/databricks/cli/libs/testserver"
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/client"
+	"github.com/databricks/databricks-sdk-go/config"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type ProxyServer struct {
+	*httptest.Server
+
+	t      testutil.TestingT
+	router *mux.Router
+
+	apiClient        *client.DatabricksClient
+	RequestCallback  func(request *testserver.Request)
+	ResponseCallback func(request *testserver.Request, response *testserver.EncodedResponse)
+}
+
+// This creates a reverse proxy server that sits in front of a real Databricks
+// workspace. This is useful for recording API requests and responses in
+// integration tests.
+//
+// Note: We cannot simply proxy the request from a localhost URL to a real
```

Comment:
>  Note: We cannot simply proxy the request from a localhost URL to a real

nit: This is a little bit confusing since we're simply proxying it (with some header modifications). Perhaps you rewrite just to mention what headers we need to add/remove/modify to make it work?

Reply:
We are not simply proxying. The SDK resolves the auth (which is pretty complex). 

>  we need to add/remove/modify to make it work?
It's not just headers, the SDK fetches OIDC tokens, figures out which token server to talk to etc.

Clarified this comment a bit more.